### PR TITLE
Adds focus state mixins and adds to components

### PR DESCRIPTION
**Pull request by macandcheese**
- Updates some tabIndex instances to make things navigable with keyboard
- Removes some lingering host-context declarations and re-adds Dir lookup (I think there might still be some lingering instances of this for Theme were we need to re-add validation since `:host-context` doesn't work in FF / Safari, I know we removed that at some point)

- Some general css cleanup

I'd imagine we should move these to `calcite-base` at some point so Calcite App Components and other consumers can use them. 

I just set up two separate mixins for now, plus a "base" you need to include to set initial values so they animate as desired. Doing this in the mixin doesn't really work since some items need to gain a focus effect when the parent `:host` is focused, etc. Open to changes there...

* Note: May need some updates to checkbox hover, switch hover, and slider positioning of numbers to accommodate these updates @bstifle @asangma 

* Note: Tree is a little weird, there are basically links contained within some of the rendered children that themselves can gain focus - @patrickarlt might have more insight on what we should do there.

* Note: Stencil also seemed to change their generated syntax for readme, so... this PR has all those since they need to be updated at some point

### CODE REVIEW

**Comment by bstifle** [comment]
image

looking pretty good man. only thing was switches and a couple other components dont have the new states, but assuming since we still need to work on those individually.

the parents on the tree might need to be addressed per the screenshot. 

nice work

**Comment by macandcheese** [comment]
OK fixed switch, yeah I mentioned tree being a little wonky in the PR description, the parent that you want to focus, can actually expand while focused and then the focus is on a group which, isn't correct...

File: src/components/calcite-pagination/calcite-pagination.tsx [hunk 1]
```diff
@@ -159,22 +161,28 @@ export class CalcitePagination {
...
 
     return pages.map(page => this.renderPage(page));
   }
 
   renderPage(num) {
     return (
-      <a class={{[CSS.page]:true, [CSS.selected]: (num === this.selectedIndex) }} onClick={() => {
-        this.selectedIndex = num;
-      }}>{num}</a>
+      <a
```
Comment (paulcpederson) [inline src/components/calcite-pagination/calcite-pagination.tsx]: for another pr, but these should probably just be buttons if they don't go anywhere. We should also probalby think about adding aria labels to these page links like `Go to page: ${num}` or something. I think screen readers are just going to read this "Internal link, 1" "Internal link, 2" "Internal link, 3"

**Comment by paulcpederson** [comment]
Niiiiice

File: src/components/calcite-pagination/calcite-pagination.tsx (hunk 1 above)
Comment (macandcheese) [reply src/components/calcite-pagination/calcite-pagination.tsx]: Agreed

**Comment by paulcpederson** [comment]
I'm going to merge this as I'd like to use them...

**Pull request merged**